crates/oxc_linter/src/rules/early_error/javascript.rs

```diff
@@ -0,0 +1,33 @@
+#[allow(clippy::wildcard_imports)]
+use oxc_ast::{ast::*, AstKind, Span};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Default, Clone)]
+pub struct EarlyErrorJavaScript;
+
+impl Rule for EarlyErrorJavaScript {
+    #[allow(clippy::single_match)]
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.get().kind() {
+            AstKind::RegExpLiteral(lit) => check_regexp_literal(lit, ctx),
+            _ => {}
+        }
+    }
+}
+
+fn check_regexp_literal(lit: &RegExpLiteral, ctx: &LintContext) {
+    #[derive(Debug, Error, Diagnostic)]
+    #[error("The 'u' and 'v' regular expression flags cannot be enabled at the same time")]
+    #[diagnostic()]
+    struct RegExpFlagUAndV(#[label] Span);
```

Comment:
Putting this inside should make things more readable once this file becomes big.